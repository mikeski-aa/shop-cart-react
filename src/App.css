body {
  background-color: #f1e5d1;
}
