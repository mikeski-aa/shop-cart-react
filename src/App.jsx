import { useState } from "react";
import "./App.css";
import { Link } from "react-router-dom";
import { Nav } from "./components/Nav";
import { getCallResults } from "./components/ApiFetch";

function App() {
  const [thisState, setThisState] = useState();

  getCallResults(setThisState);
  return (
    <div>
      <h1>Homepage</h1>
      <Nav />
      <div role="test" className="lol">
        dddd
      </div>
      <div>{state}</div>
    </div>
  );
}

export default App;
