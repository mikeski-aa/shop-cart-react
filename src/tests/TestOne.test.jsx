import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import App from "../App.jsx";

describe("testing whether the thing works", () => {
  it("true to be true", () => {
    expect(true).toBe(true);
  });

  it("expect false to be false", () => {
    expect(false).toBe(false);
  });
});

describe("App component", () => {
  it("renders correct heading", () => {
    render(<App />);
    expect(screen.getByRole("heading").textContent).toMatch("Vite + React");
  });

  it("renders", () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
