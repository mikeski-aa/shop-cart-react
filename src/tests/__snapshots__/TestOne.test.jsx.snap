// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders 1`] = `
<div>
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="/vite.svg"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button>
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.jsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;
